src/main/kotlin/agallipo/BoardingCard.kt

```diff
@@ -1,59 +1,59 @@
-package agallipo
+    package agallipo
 
-import java.io.File
-import java.io.InputStream
+    import java.io.File
+    import java.io.InputStream
+    //created by alex :)
+    data class BoardingCard(var origin: String?, var destination: String?,  var transport: String?, var gate: String?, var seat: String?)
+    //data class CardInformation(val flight: String?, val gate: String?, val seat: String?)
 
-data class BoardingCard(var origin: String?, var destination: String?,  var transport: String?, var gate: String?, var seat: String?)
-//data class CardInformation(val flight: String?, val gate: String?, val seat: String?)
 
-
-interface BoardingCardsReader {
-    fun readCards() : MutableList<String>
-}
-
-class BoardingCardsFile(val pathname: String) : BoardingCardsReader {
-    override fun readCards(): MutableList<String> {
-        val inputStream: InputStream = File(pathname).inputStream()
-        val lines = mutableListOf<String>()
-        inputStream.bufferedReader().forEachLine(lines::add)
-        return lines
+    interface BoardingCardsReader {
+        fun readCards() : MutableList<String>
     }
-}
-
-interface BoardingCardClassifier {
-    fun parseLine(boardingCardsFile: BoardingCardsFile) : MutableList<BoardingCard?> {
-        var boardingCards = mutableListOf<BoardingCard?>()
-        var boardingCard: BoardingCard? = BoardingCard(null,null,null,null,null)
-        var lines = boardingCardsFile.readCards()
-
-        for (line in lines ) {
-            if (line.compareTo("{") != 0 && line.compareTo("}") != 0)
-               setCardValues(line, boardingCard)
-            if (line.compareTo("}") == 0)
-                boardingCards.add(boardingCard!!.copy())
+
+    class BoardingCardsFile(val pathname: String) : BoardingCardsReader {
+        override fun readCards(): MutableList<String> {
+            val inputStream: InputStream = File(pathname).inputStream()
+            val lines = mutableListOf<String>()
+            inputStream.bufferedReader().forEachLine(lines::add)
+            return lines
         }
-        return boardingCards
     }
 
-    fun setCardValues(line: String, boardingCard: BoardingCard?){
+    interface BoardingCardClassifier {
```

Comment:
Why an interface here if you implements all the functions inside it?

Reply:
In my first jira ticket I was asked to make an interface with all the methods which gets all the boarding card information from the JSON, having a function called GetAllCards. I assumed then that an interface could have a specific task (like getting the information of the cards or sorting them) which is developed in different functions of itself.

Should I make it a class?

Reply:
Now that I'm using the Gson library I deleted this interface, but that was a bad use for an interface. Now is solved.